### PR TITLE
[OP-41] Refactored data providers.

### PR DESCRIPTION
**Pull request by mark-terry**
Refactored the data providers and other required code.

### CODE REVIEW

File: src/components/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -13,26 +15,29 @@ import { ToastProvider } from "../../context/toasts";
...
-                <EnodeTab isOpen={tab === ENODE_TAB} />
+                <AdminDataProvider>
+                    <AccountDataProvider>
+                        <AccountTab isOpen={tab === ACCOUNT_TAB} />
+                    </AccountDataProvider>
+                    <AdminTab isOpen={tab === ADMIN_TAB} />
+                    <NodeDataProvider>
+                        <EnodeTab isOpen={tab === ENODE_TAB} />
+                    </NodeDataProvider>
+                </AdminDataProvider>
```
Comment (usmansaleem) [inline src/components/Dashboard/Dashboard.tsx]: I am assuming it is by design? i.e. AccountDataProvider is nested under AdminDataProvider?

Comment (Errorific) [reply src/components/Dashboard/Dashboard.tsx]: yeah, admin data is need in all 3 tabs to see if we're the admin, account & node data is only needed in their respective tabs

File: src/containers/Tabs/Account.tsx [hunk 2]
```diff
@@ -98,7 +101,10 @@ const AccountTabContainer: React.FC<AccountTabContainerProps> = ({ isOpen }) =>
             });
     };
 
-    return (
+    return isOpen ?
```
Comment (Errorific) [inline src/containers/Tabs/Account.tsx]: even though I was on zoom with you when we wrote this, need to refactor it to make it readable.
Maybe ```
const allDataReady = dataReady && adminDataReady
if (isOpen && !allDataReady) {
  return <LoadingPage />
} else if (isOpen && allDataReady) {
  return ...
} else if (!isOpen) {
  return ...
}```

File: src/containers/Tabs/Admin.js [hunk 3]
```diff
@@ -94,19 +94,25 @@ const AdminTabContainer = ({ isOpen }) => {
...
-            modals={modals}
-            toggleModal={toggleModal}
-            handleAdd={handleAdd}
-            handleRemove={handleRemove}
-            isAdmin={isAdmin}
-            deleteTransaction={deleteTransaction}
-            isValid={isAddress}
-            isOpen={isOpen}
-        />
+    return isOpen ? (
```
Comment (Errorific) [inline src/containers/Tabs/Admin.js]: Same as previous nested ternary, needs refactoring

File: src/containers/Tabs/Enode.js [hunk 4]
```diff
@@ -168,22 +167,28 @@ const EnodeTabContainer = ({ isOpen }) => {
...
-            handleRemove={handleRemove}
-            handleLock={handleLock}
-            isAdmin={isAdmin}
-            deleteTransaction={deleteTransaction}
-            isValid={isValidEnode}
-            pendingLock={!!transactions.get("lock")}
-            isReadOnly={isReadOnly}
-            isOpen={isOpen}
-        />
+    return isOpen ? (
```
Comment (Errorific) [inline src/containers/Tabs/Enode.js]: More nested ternary

File: src/context/nodeData.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        nodeWhitelist,
+        setNodeWhitelist
+    ]);
+    return <DataContext.Provider value={value} {...props} />;
+};
+
+export const useNodeData = () => {
+    const context = useContext(DataContext);
+    if (!context) {
+        throw new Error("useData must be used within a DataProvider.");
```
Comment (Errorific) [inline src/context/nodeData.tsx]: error text s/useData/useNodeData/

Comment (Errorific) [reply src/context/nodeData.tsx]: also s/DataProvider/NodeDataProvider/

File: src/context/adminData.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        setAdmins
+    ])
+    return <AdminDataContext.Provider value={value} {...props} />
+}
+
+export const useAdminData = () => {
+    const context = useContext(AdminDataContext)
+
+    if (!context) {
+        throw new Error("useAccountData must be used within a AccountDataProvider.")
```
Comment (Errorific) [inline src/context/adminData.tsx]: s/useAccountData/useAdminData/  s/AccountDataProvider/AdminDataProvider/

**Pull request merged**